src/array/array.cpp

```diff
@@ -3,41 +3,48 @@
 
 namespace aero {
 
-  Array::Array(std::size_t size) :
-    values_(size, 0.0) {}
+Array::Array(std::size_t size)
+  : values_(size, 0.0) {}
 
-  Array:: Array(std::size_t size,
-      Real initial_value) : values_(size, initial_value) {}
+Array:: Array(std::size_t size, Real initial_value)
+  : values_(size, initial_value) {}
 
-  Array::Array(const std::vector<Real> &values) : values_(values) {}
+Array::Array(const std::vector<Real> &values)
+  : values_(values) {}
 
-  Array& Array::operator=(const std::vector<Real> &values) {
-    this->values_ = values;
-    return *this;
-  }
+Array& Array::operator=(const std::vector<Real> &values) {
+  this->values_ = values;
+  return *this;
+}
 
-  Array* Array::clone() const {
-    return new Array(*this);
-  }
+Array* Array::clone() const {
+  return new Array(*this);
+}
 
-  void Array::copy_in(const Real *input) {
-    for (int i=0; i<this->values_.size(); ++i) this->values_[i] = input[i];
+void Array::copy_in(const Real *input) {
+  for (int i=0; i<this->values_.size(); ++i) {
+    this->values_[i] = input[i];
   }
+}
 
-  void Array::copy_in(const std::vector<Real> &input) {
-    this->copy_in(input.data());
-  }
+void Array::copy_in(const std::vector<Real> &input) {
+  this->values_.resize(input.size());
+  this->copy_in(input.data());
+}
 
-  void Array::copy_out(Real *output) const {
-    for (int i=0; i<this->values_.size(); ++i) output[i] = this->values_[i];
+void Array::copy_out(Real *output) const {
+  for (int i=0; i<this->values_.size(); ++i) {
+    output[i] = this->values_[i];
   }
+}
 
-  void Array::copy_out(std::vector<Real> &output) const {
-    this->copy_out(output.data());
-  }
+void Array::copy_out(std::vector<Real> &output) const {
+  output.resize(this->size());
+  this->copy_out(output.data());
+}
```

Comment:
I think this is actually a bit different, since we have the ability to prevent an error here without incurring a cost if it's not needed. But maybe we could `assert` that the output vector has the proper size instead.

Reply:
the `assert` is also something we could implement in every language

Reply:
Side note: we probably need to make our own `AERO_ASSERT` macro that is MPI-aware.

Reply:
I like that - then we can also use the same syntax for asserts across languages

Reply:
I added an `AERO_ASSERT` macro for C/C++. How do you think we should do this in Fortran? Should we have some sort of generated `.inc` file available (built from, say, `include/aero/aero.inc.in`) for these kinds of macros?